test/benchmarks/material/table/app.module.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {A11yModule} from '@angular/cdk/a11y';
+import {Component, NgModule, ViewEncapsulation} from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
+import {BasicTable, GenericObject} from './basic-table';
+import {MatTableModule} from '@angular/material/table';
+
+/**
+ * @title Checkbox benchmark component.
+ */
+@Component({
+  selector: 'app-root',
+  template: `
+    <button id="show" (click)="show()">Show</button>
+    <button id="hide" (click)="hide()">Hide</button>
+
+    <button id="ten-rows" (click)="updateTable({ numRows: 10 })">10 Rows</button>
+    <button id="one-hundred-rows" (click)="updateTable({ numRows: 100 })">100 Rows</button>
+    <button id="one-thousand-rows" (click)="updateTable({ numRows: 1000 })">1000 Rows</button>
+
+    <button id="five-cols" (click)="updateTable({ numCols: 5 })">5 Cols</button>
+    <button id="ten-cols" (click)="updateTable({ numCols: 10 })">10 Cols</button>
+    <button id="twenty-cols" (click)="updateTable({ numCols: 20 })">20 Cols</button>
+
+    <ng-container *ngIf="isVisible">
+      <basic-table [rows]="createRows()" [cols]="createCols()"></basic-table>
```

Comment:
I would avoid binding to the function like this, because the result is going to be different every time it's evaulated because it's a different array reference. I would create constants at the file level